addons/move-to-top-layer/userscript.js

```diff
@@ -1,15 +1,37 @@
 // Initial code was written by Norbiros
 
-export default async function ({ addon, console }) {
+export default async function ({ addon, console, msg }) {
   const vm = addon.tab.traps.vm;
+
+  function spriteToFront(name) {
+    const target = vm.runtime.getSpriteTargetByName(name);
+    target.goToFront();
+    target.setVisible(true);
+  }
+
   document.body.addEventListener("click", (e) => {
     if (e.shiftKey && !addon.self.disabled) {
       const parentDiv = e.target.closest("div[class^='sprite-selector_sprite-wrapper']");
       if (parentDiv) {
+        if (
+          e.target.closest("div[class^='delete-button_delete-button_']") ||
+          e.target.closest(".sa-sprite-properties-info-btn")
+        )
+          return;
         const spriteName = parentDiv.querySelector("div[class^='sprite-selector-item_sprite-name']").innerText;
-        // move the sprite with that name to front
-        vm.runtime.getSpriteTargetByName(spriteName).goToFront();
+        spriteToFront(spriteName);
       }
     }
   });
+  addon.tab.createEditorContextMenu(
+    (ctx) => {
+      spriteToFront(ctx.name);
```

Comment:
This doesn't work if `folders` is enabled because it replaces the sprite's name with an object. The best way to fix this is probably in the addon API: [here](https://github.com/ScratchAddons/ScratchAddons/blob/833fff4/addon-api/content-script/contextmenu.js#L94), `props.name.realName` should be used instead of `props.name` if it exists.

Reply:
How should I handle compatibility when shift-clicking?

Reply:
One way - I don't know if it's the simplest - is to read the name from the React component props instead of the DOM.